File: src/lib/analytics.ts
import * as api from "@/lib/tauri";

export async function trackEvent(
  eventName: string,
  properties: Record<string, unknown> = {}
): Promise<void> {
  try {
    await api.trackEvent(eventName, properties);
  } catch (error) {
    console.warn("analytics tracking failed", error);
  }
}

File: src-tauri/src/commands/analytics.rs
use crate::db;
use crate::error::AppError;
use crate::state::AppState;
use serde_json::Value;
use tauri::State;

#[tauri::command]
pub fn track_event(
    state: State<AppState>,
    event_name: String,
    properties: Option<Value>,
) -> Result<(), AppError> {
    let conn = state.db.lock().unwrap();
    let payload = properties.unwrap_or_else(|| serde_json::json!({}));
    db::analytics::track_event(&conn, &event_name, &payload)?;
    Ok(())
}
